builtInFunctions/esdtNFTCreate.go

```diff
@@ -112,17 +119,42 @@ func (e *esdtNFTCreate) ProcessBuiltinFunction(
 	if err != nil {
 		return nil, err
 	}
-	if len(vmInput.Arguments) < 7 {
+
+	minNumOfArgs := 7
+	if vmInput.CallType == vm.ExecOnDestByCaller {
+		minNumOfArgs = 8
+	}
+	lenArgs := len(vmInput.Arguments)
+	if lenArgs < minNumOfArgs {
 		return nil, fmt.Errorf("%w, wrong number of arguments", ErrInvalidArguments)
 	}
 
+	accountWithRoles := acntSnd
+	uris := vmInput.Arguments[6:]
+	if vmInput.CallType == vm.ExecOnDestByCaller {
+		scAddressWithRoles := vmInput.Arguments[lenArgs-1]
+		uris = vmInput.Arguments[6 : lenArgs-1]
+
+		if len(scAddressWithRoles) != len(vmInput.CallerAddr) {
+			return nil, ErrInvalidAddressLength
+		}
+		if bytes.Equal(scAddressWithRoles, vmInput.CallerAddr) {
```

Comment:
!byte.Equal

Reply:
the current condition is correct. so you should not put the same address.

Reply:
discussed, the vmInput.CallType == vm.ExecOnDestByCaller comes directly from the VM. So the VM is responsible for setting the last argument to the correct value. The protocol trusts the VM